clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -2926,7 +2926,7 @@ static bool checkWorkGroupSizeValues(Sema &S, Decl *D, const ParsedAttr &Attr,
   }
 
   if (const auto *A = D->getAttr<SYCLIntelMaxGlobalWorkDimAttr>())
-    if (A->getNumber() == 0)
+    if (A->getValue() == 0)
```

Comment:
I guess to fix missing diagnostic you need to attribute access value the same way as you do it in CodeGen, i.e.
```suggestion
    int64_t AttrValue = A->getValue()->getIntegerConstantExpr(S.Context)->getSExtValue();
    if (AttrValue == 0)
```

Reply:
Done